code/modules/mob/living/basic/space_fauna/morph.dm

```diff
@@ -0,0 +1,215 @@
+
+	med_hud_set_health()
+	med_hud_set_status() //we're an object honest
+
+	visible_message(
+		span_warning("[src] suddenly twists and changes shape, becoming a copy of [target]!"),
+		span_notice("You twist your body and assume the form of [target]."),
+	)
+
+	form_weakref = WEAKREF(target)
+	form_typepath = target.type
+
+/// Do some more logic for the morph when we undisguise through the action.
+/mob/living/basic/morph/proc/on_undisguise()
+	SIGNAL_HANDLER
+	visible_message(
+		span_warning("[src] suddenly collapses in on itself, dissolving into a pile of green flesh!"),
+		span_notice("You reform to your normal body."),
+	)
+
+	//Baseline stats
+	melee_damage_lower = initial(melee_damage_lower)
+	melee_damage_upper = initial(melee_damage_upper)
+	remove_movespeed_modifier(/datum/movespeed_modifier/morph_disguised)
+
+	med_hud_set_health()
+	med_hud_set_status() //we are no longer an object
+
+	form_weakref = null
+	form_typepath = null
+
+/// Alias for the disguise ability to be used as a keybind.
+/mob/living/basic/morph/proc/trigger_ability(mob/living/basic/source, atom/target)
+	SIGNAL_HANDLER
+
+	// linters hate this if it's not async for some reason even though nothing blocks
+	INVOKE_ASYNC(disguise_ability, TYPE_PROC_REF(/datum/action/cooldown, InterceptClickOn), caller = source, target = target)
+	return COMSIG_MOB_CANCEL_CLICKON
+
+/// Handles the logic for attacking anything.
+/mob/living/basic/morph/proc/pre_attack(mob/living/basic/source, atom/target)
+	SIGNAL_HANDLER
+
+	if(HAS_TRAIT(src, TRAIT_DISGUISED) && (melee_damage_disguised >= 0))
+		balloon_alert(src, "can't attack while disguised!")
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+	if(isliving(target)) //Eat Corpses to regen health
+		var/mob/living/living_target = target
+		if(living_target.stat != DEAD)
+			return
+
+		INVOKE_ASYNC(source, PROC_REF(eat), eatable = living_target, delay = 3 SECONDS, update_health = -50)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+	if(isitem(target)) //Eat items just to be annoying
+		var/obj/item/item_target = target
+		if(item_target.anchored)
+			return
+
+		INVOKE_ASYNC(source, PROC_REF(eat), eatable = item_target, delay = 2 SECONDS)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+/// Eat stuff. Delicious. Return TRUE if we ate something, FALSE otherwise.
+/// Required: `eatable` is the thing (item or mob) that we are going to eat.
+/// Optional: `delay` is the applicable time-based delay to pass into `do_after()` before the logic is ran.
+/// Optional: `update_health` is an integer that will be added (or maybe subtracted if you're cruel) to our health after we eat something. Passed into `adjust_health()` so make sure what you pass in is accurate.
+/mob/living/basic/morph/proc/eat(atom/movable/eatable, delay = 0 SECONDS, update_health = 0)
+	if(QDELETED(eatable) || eatable.loc == src)
+		return FALSE
+
+	if(HAS_TRAIT(src, TRAIT_DISGUISED) && !eat_while_disguised)
+		to_chat(src, span_warning("You cannot eat anything while you are disguised!"))
```

Comment:
Balloon alert? Or do you think this is worth maintaining a log for the player?

Reply:
log probably unneeded